### PR TITLE
Fix build CI job by updating macOS images

### PR DESCRIPTION
**Pull request by fluiddot**
## Proposed Changes

- Use different macOS images to ensure that we build `x86` and `arm64` architectures:
  - [`macos-13`](https://github.com/actions/runner-images/tree/main?tab=readme-ov-file#available-images): `x86` Intel
  - [`macos-14`](https://github.com/actions/runner-images/tree/main?tab=readme-ov-file#available-images): `arm64` Apple silicon

## Testing Instructions

- Observe that all CI jobs succeed.

## Pre-merge Checklist

- [x] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: .github/workflows/build-installer.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - trunk
+      - fix/ci-macos-build-installers # Added only for testing purpose
```
Comment (fluiddot) [inline .github/workflows/build-installer.yml:7]: I'll remove this once the PR is approved.

**Comment by wojtekn** [comment]
@fluiddot as we build on Buildkite now, do we still need the same on GH actions?

**Comment by fluiddot** [comment]
> @fluiddot as we build on Buildkite now, do we still need the same on GH actions?

@wojtekn As far as I know, Buildkite only produces builds for Git tags, not PRs. We could remove the GitHub actions but it might be interesting to keep them to check that PRs don't break the build process with failures. WDYT?

**Comment by wojtekn** [comment]
@fluiddot I see, makes sense.

**Comment by fluiddot** [comment]
In the future, it would be great if the build jobs were run by Buildkite.

**Review by katinthehatsite — APPROVED** [review]
I hope I am checking this correctly. This is what it looks on my end:

Screenshot 2024-04-25 at 4 34 36 PM

In that case, it seems that everything succeeds! 👍

**Comment by fluiddot** [comment]
> I hope I am checking this correctly. This is what it looks on my end:
> 
> Screenshot 2024-04-25 at 4 34 36 PM
> In that case, it seems that everything succeeds! 👍

Yes, we only need to test that `Build Installer` CI jobs pass. Thanks @katinthehatsite for reviewing the PR 🙇 !

**Pull request merged**